spectral_cube/stokes_spectral_cube.py

```diff
@@ -0,0 +1,201 @@
+from __future__ import print_function, absolute_import, division
+
+import numpy as np
+
+from astropy.extern import six
+from .spectral_cube import SpectralCube
+from . import wcs_utils
+from .masks import BooleanArrayMask, is_broadcastable_and_smaller
+
+__all__ = ['StokesSpectalCube']
+
+VALID_STOKES = ['I', 'Q', 'U', 'V', 'RR', 'LL', 'RL', 'LR']
+
+
+class StokesSpectralCube(object):
+    """
+    A class to store a spectral cube with multiple Stokes parameters.
+
+    The individual Stokes cubes can share a common mask in addition to having
+    component-specific masks.
+    """
+
+    def __init__(self, stokes_data, mask=None, meta=None, fill_value=None):
+
+        self._stokes_data = stokes_data
+        self._meta = meta or {}
+        self._fill_value = fill_value
+
+        reference = tuple(stokes_data.keys())[0]
+
+        for component in stokes_data:
+
+            if not isinstance(stokes_data[component], SpectralCube):
+                raise TypeError("stokes_data should be a dictionary of "
+                                "SpectralCube objects")
+
+            if not wcs_utils.check_equality(stokes_data[component].wcs,
+                                            stokes_data[reference].wcs):
+                raise ValueError("All spectral cubes in stokes_data "
+                                 "should have the same WCS")
+
+            if component not in VALID_STOKES:
+                raise ValueError("Invalid Stokes component: {0} - should be "
+                                 "one of I, Q, U, V, RR, LL, RL, LR".format(component))
+
+            if stokes_data[component].shape != stokes_data[reference].shape:
+                raise ValueError("All spectral cubes should have the same shape")
+
+        self._wcs = stokes_data[reference].wcs
+        self._shape = stokes_data[reference].shape
+
+        if isinstance(mask, BooleanArrayMask):
+            if mask.shape != self._shape:
```

Comment:
Instead, should use `if ~is_broadcastable_and_smaller(mask.shape, self._shape):`